sdk/storage/src/core/perform_request_response.rs

```diff
@@ -0,0 +1,23 @@
+use hyper::client::ResponseFuture;
+use url::Url;
+
+#[derive(Debug)]
+pub struct PerformRequestResponse {
+    pub(crate) url: Url,
+    pub(crate) response_future: ResponseFuture,
+}
+
+impl PerformRequestResponse {
+    pub fn url(&self) -> &Url {
+        &self.url
+    }
+}
+
+impl std::convert::From<(Url, ResponseFuture)> for PerformRequestResponse {
```

Comment:
Not sure I like this. I think I'd prefer a `PerformRequestResponse::new` function. Tuples are best for data that is extremely obviously related to one another, which is not the case here. The new function will be a bit easier to understand.